test/test_config.py

```diff
@@ -0,0 +1,155 @@
+#
+# Shoulder
+# Copyright (C) 2018 Assured Information Security, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import unittest
+import os
+import sys
+
+from shoulder.config import Configuration
+from shoulder.config import ShoulderConfig
+from test.support.constants import *
+
+class TestConfiguration(unittest.TestCase):
+
+    def test_configuration_init(self):
+        c = Configuration("test_name", "test_val")
+        self.assertIsNotNone(c)
+        self.assertEqual(c.name, "test_name")
+        self.assertIsNotNone(c.default_value, "test_val")
+
+    def test_configuration_str(self):
+        c = Configuration("test_name", "test_val")
+        self.assertTrue(str(c))
+        c.options = ["test_val", "test_val2"]
+        self.assertTrue(str(c))
+
+    def test_configuration_help(self):
```

Comment:
There's `io.StringIO` if you want to actually capture and inspect the output. I'm not sure how useful it is to just run it into `/dev/null` and see that it doesn't crash?

Reply:
I think this test case is kinda silly in the first place, it exists purely to get test coverage to 100%. From a unit-test perspective, I don't think I really care what the output looks like, just that Configuration.help() doesn't throw any exceptions.

Reply:
```
def test_configuration_help(self):
    pass
```

:imp:

Reply:
Hahaha, well its not quite *that* useless. At least the test thats there would warn if you got rid of that method, or it tries to print an attribute that doesn't exist.